docs/Advanced-Topics-Custom-Block-Render.md

```diff
@@ -0,0 +1,148 @@
+
+When pasting content or when using the
+[convertFromHTML](https://facebook.github.io/draft-js/docs/api-reference-data-conversion.html#convertfromhtml)
+Draft will then convert the pasted content to the respective block rendering type
+by matching the Draft block render map with the matched tag.
+
+## Draft default block render map
+
+| HTML element    | Draft block type                        |
+| --------------- | --------------------------------------- |
+| `<h1/>`         | header-one                              |
+| `<h2/>`         | header-two                              |
+| `<h3/>`         | header-three                            |
+| `<h4/>`         | header-four                             |
+| `<h5/>`         | header-five                             |
+| `<h6/>`         | header-six                              |
+| `<h6/>`         | header-six                              |
+| `<blockquote/>` | blockquote                              |
+| `<pre/>`        | code-block                              |
+| `<figure/>`     | atomic                                  |
+| `<li/>`         | unordered-list-item,ordered-list-item** |
+| `<div/>`        | unstyled***                             |
+
+\*\* - Block type will be based on the parent `<ul/>` or `<ol/>`
+
+\*\*\* -  Any block that is not recognized by the block redering mapping will be treated as unstyled
+
+## Configuring block render map
+
+Draft default block render map can be overwritten, by passing an
+[Immutable Map](http://facebook.github.io/immutable-js/docs/#/Map) to
+the editor blockRender props.
+
+*example of overwritting default block render map:*
+
+```js
+// The example below deliberatly only allows
+// 'heading-two' and 'paragraph' as valid block types and
+// updated the unstyled element to become a paragraph.
+const blockRenderMap = Immutable.Map({
+	'header-two': {
+		element: 'h2'
+	},
+	'paragraph': {
+		element: 'p'
+	},
+	'unstyled': {
+		element: 'p'
+	}
+});
+
+class RichEditor() {
+    ...
+    render() {
+      return (<Editor ...
+                blockRenderMap={blockRenderMap} />);
+    }
+}
+```
+
+There are cases where instead of overwriting the defaults we just want to add new block types;
+this can be done by using the DefaultDraftBlockRenderMap reference to create a new blockRenderMap
+
+*example of extending default block render map:*
+
+```js
+const blockRenderMap = Immutable.Map({
+	'paragraph': {
+		element: 'p'
+	},
+	'unstyled': {
+		element: 'p'
+	}
+});
+
+// Include 'paragraph' as a valid block and updated the unstyled element but
+// keep support for other draft default block types
+const extendedBlockRenderMap = Draft.DefaultDraftBlockRenderMap.merge(blockRenderMap);
+
+class RichEditor() {
+    ...
+    render() {
+      return (<Editor ...
+                blockRenderMap={extendedBlockRenderMap} />);
+    }
+}
+```
+
+## Custom block wrappers
+
+By default the html element is used to wrap block types however a react component
```

Comment:
This functionality was already there, but your idea make sense. maybe could be included on a PR.